base_layer/core/src/transactions/transaction.rs

```diff
@@ -223,6 +231,39 @@ impl UnblindedOutput {
         }
         Ok(output)
     }
+
+    pub fn as_rewindable_transaction_output(
+        &self,
+        factories: &CryptoFactories,
+        rewind_data: &RewindData,
+    ) -> Result<TransactionOutput, TransactionError>
+    {
+        let commitment = factories.commitment.commit(&self.spending_key, &self.value.into());
+
+        let proof_bytes = factories.range_proof.construct_proof_with_rewind_key(
+            &self.spending_key,
+            self.value.into(),
+            &rewind_data.rewind_key,
+            &rewind_data.rewind_blinding_key,
+            &rewind_data.proof_message,
+        )?;
+
+        let proof = RangeProof::from_bytes(&proof_bytes)
+            .map_err(|_| TransactionError::RangeProofError(RangeProofError::ProofConstructionError))?;
+
+        let output = TransactionOutput {
+            features: self.features.clone(),
+            commitment,
+            proof,
+        };
+        // A range proof can be constructed for an invalid value so we should confirm that the proof can be verified.
+        if !output.verify_range_proof(&factories.range_proof)? {
```

Comment:
Might be worth skipping this in Release mode. Is it not cheaper to verify `self.value` directly or is there another way of creating an invalid rangeproof?

Reply:
🤷‍♂️  This is from the original implementation of UnblindedOutput. If we are confident that the value in self.value cannot be invalid I guess we can remove this here and in 'as_transaction_output`